v2/pkg/engine/datasource/httpclient/nethttpclient.go

```diff
@@ -244,45 +269,46 @@ func makeHTTPRequest(client *http.Client, ctx context.Context, url, method, head
 	}
 	trace, err := json.Marshal(responseTrace)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	responseWithTraceExtension, err := jsonparser.Set(data, trace, "extensions", "trace")
 	if err != nil {
-		return err
+		return nil, err
 	}
-	_, err = out.Write(responseWithTraceExtension)
-	return err
+	return responseWithTraceExtension, nil
 }
 
-func Do(client *http.Client, ctx context.Context, requestInput []byte, out *bytes.Buffer) (err error) {
+func Do(client *http.Client, ctx context.Context, baseHeaders http.Header, requestInput []byte) (data []byte, err error) {
 	url, method, body, headers, queryParams, enableTrace := requestInputParams(requestInput)
-	h := pool.Hash64.Get()
-	_, _ = h.Write(body)
-	bodyHash := h.Sum64()
-	pool.Hash64.Put(h)
-	ctx = context.WithValue(ctx, bodyHashContextKey{}, bodyHash)
-	return makeHTTPRequest(client, ctx, url, method, headers, queryParams, bytes.NewReader(body), enableTrace, out, ContentTypeJSON)
+	return makeHTTPRequest(client, ctx, baseHeaders, url, method, headers, queryParams, bytes.NewReader(body), enableTrace, ContentTypeJSON, len(body))
 }
 
 func DoMultipartForm(
-	client *http.Client, ctx context.Context, requestInput []byte, files []*FileUpload, out *bytes.Buffer,
-) (err error) {
+	client *http.Client, ctx context.Context, baseHeaders http.Header, requestInput []byte, files []*FileUpload,
+) (data []byte, err error) {
 	if len(files) == 0 {
-		return errors.New("no files provided")
+		return nil, errors.New("no files provided")
 	}
 
 	url, method, body, headers, queryParams, enableTrace := requestInputParams(requestInput)
 
-	h := pool.Hash64.Get()
-	defer pool.Hash64.Put(h)
-	_, _ = h.Write(body)
-
 	formValues := map[string]io.Reader{
 		"operations": bytes.NewReader(body),
 	}
 
 	var tempFiles []*os.File
 
+	defer func() {
+		for _, file := range tempFiles {
+			if err := file.Close(); err != nil {
+				continue
+			}
+			if err = os.Remove(file.Name()); err != nil {
```

Comment:
I suggest always removing the file regardless of whether it can be closed. A close error might not always point to a non-existent file.

```go
_ = file.Close()
_ = os.Remove(file.Name())
```